opentimelineio/schema/gap.py

```diff
@@ -32,6 +35,36 @@ class Gap(core.Item):
     _serializable_label = "Gap.1"
     _class_path = "schema.Gap"
 
+    def __init__(
+        self,
+        name=None,
+        # note - only one of the following is accepted
+        duration=None,
+        source_range=None,
+        effects=None,
+        markers=None,
+        metadata=None,
+    ):
+        if duration and source_range:
+            raise RuntimeError(
+                "Cannot instantiate with both a source range and a duration."
+            )
```

Comment:
We should also check for neither and complain.

Reply:
Unfortunately, the deserializer requires that we have the ability to `__init__` classes without any arguments, so we can't make this an error case.